### PR TITLE
Em 93

### PR DESCRIPTION
**Pull request by idan-matrix**

### CODE REVIEW

File: src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js [hunk 1]
```diff
@@ -143,6 +191,156 @@ const DiagnosisAndRecommendations = ({
...
+            medicationRequest['recorder'] = store.getState().login.userID;
+            medicationRequest['note'] = drug.instructionsForTheDrug;
+            medicationRequest['routeCode'] = drug.drugRoute;
+            medicationRequest['medicationCodeableConceptCode'] = drug.drugName;
+            medicationRequest['timingCode'] = drug.intervals;
+            medicationRequest['doseQuantity'] = drug.quantity;
+            medicationRequest['methodCode'] = drug.drugForm;
+            medicationRequest['timingRepeatStart'] = moment(
+              drug.toDate,
+              'DD/MM/YYYY',
```
Comment (matrix-amiel) [inline src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js:301]: use format date from redux

File: src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js [hunk 2]
```diff
@@ -143,6 +191,156 @@ const DiagnosisAndRecommendations = ({
...
+            )
+              .subtract(drug.duration, 'days')
+              .format('YYYY-MM-DD');
+            medicationRequest['timingRepeatEnd'] = moment(
+              drug.toDate,
+              'DD/MM/YYYY',
```
Comment (matrix-amiel) [inline src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js:307]: use format date

File: src/Components/Forms/DiagnosisAndRecommendations/DrugRecommendation/index.js [hunk 3]
```diff
@@ -58,6 +62,223 @@ const DrugRecommendation = () => {
...
+                  [`drugRecommendation[${
+                    medicationRequestIndex - 1
+                  }].duration`]: duration,
+                },
+                {
+                  [`drugRecommendation[${medicationRequestIndex - 1}].toDate`]:
+                    moment(
+                      normalizedFhirMedicationRequest.timingRepeatEnd,
+                      'YYYY-MM-DD',
+                    ).format('DD/MM/YYYY') || '',
```
Comment (matrix-amiel) [inline src/Components/Forms/DiagnosisAndRecommendations/DrugRecommendation/index.js:222]: use format date from redux

File: src/Components/Forms/DiagnosisAndRecommendations/DrugRecommendation/index.js [hunk 4]
```diff
@@ -58,6 +62,223 @@ const DrugRecommendation = () => {
...
+                quantity: normalizedFhirMedicationRequest.doseQuantity || '',
+                drugForm: normalizedFhirMedicationRequest.methodCode || '',
+                drugRoute: normalizedFhirMedicationRequest.routeCode || '',
+                intervals: normalizedFhirMedicationRequest.timingCode || '',
+                duration,
+                toDate:
+                  moment(
+                    normalizedFhirMedicationRequest.timingRepeatEnd,
+                    'YYYY-MM-DD',
+                  ).format('DD/MM/YYYY') || '',
```
Comment (matrix-amiel) [inline src/Components/Forms/DiagnosisAndRecommendations/DrugRecommendation/index.js:244]: format date from redux

File: src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js [hunk 5]
```diff
@@ -143,6 +191,156 @@ const DiagnosisAndRecommendations = ({
...
+                  source: patient.id,
+                  item: items,
+                },
+              },
+            }),
+          );
+        }
+        //Updating encounter
+        const cloneEncounter = { ...encounter };
+        cloneEncounter.extensionSecondaryStatus = data.nextStatus;
```
Comment (matrix-amiel) [inline src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js:271]: the encounter status need to update only when the saving occurs on click a save button and not in the when move between tabs, how is it implemented here?

File: src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js [hunk 6]
```diff
@@ -143,6 +191,156 @@ const DiagnosisAndRecommendations = ({
...
+                FHIR('MedicationRequest', 'doWork', {
+                  functionName: 'createMedicationRequest',
+                  functionParams: {
+                    medicationRequest,
+                  },
+                }),
+              );
+            }
+          });
+          const res = await Promise.all[APIsArray];
```
Comment (matrix-amiel) [inline src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js:334]: I think the onSubmit function needs to be in the GeneralComponents/SaveForm/index.js component.
the SaveForm need to get in the props function with all the logic of the validation and saving to a server. in the onsubmit in the  SaveForm component we will do 3 steps:
1. run the function from the parent
2. update encounter
3. move to patient tracking

when you move between tabs you dont need to implemt submit but only the function that configured in the parent.

File: src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndTreatment/index.js [hunk 7]
```diff
@@ -41,10 +42,38 @@ const DiagnosisAndTreatment = () => {
...
-
   const treatmentDetails = t('Treatment details');
   const diagnosisDetails = t('Diagnosis details');
   const findingsDetails = t('Findings details');
+  React.useEffect(() => {
+    const { items } = questionnaireResponse;
+    if (items) {
+      items.forEach((item) => {
+        if (item.answer) {
+          switch (item.text) {
```
Comment (matrix-amiel) [inline src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndTreatment/index.js:53]: the switch needs to be on the linkId, the text can be changed. it's better to put the text in a comment then the code will be more readable

File: src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js [hunk 8]
```diff
@@ -143,6 +191,156 @@ const DiagnosisAndRecommendations = ({
...
+  const onSubmit = async (data) => {
+    if (isRequiredValidation(data)) {
+      try {
+        const APIsArray = [];
+        const items = qItem.item.map((i) => {
+          const item = {
+            linkId: i.linkId,
+            text: i.text,
+          };
+          switch (i.text) {
```
Comment (matrix-amiel) [inline src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js:204]: the switch needs to be on the linkId, the text can be changed. it's better to put the text in a comment then the code will be more readable

File: src/Components/Forms/DiagnosisAndRecommendations/DecisionOnRelease/index.js [hunk 9]
```diff
@@ -31,8 +32,30 @@ const RecommendationsOnRelease = () => {
   useEffect(() => {
     register({ name: 'decision' });
     register({ name: 'evacuationWay' });
+    const { items } = questionnaireResponse;
+    if (items) {
+      items.forEach((item) => {
+        if (item.answer) {
+          switch (item.text) {
```
Comment (matrix-amiel) [inline src/Components/Forms/DiagnosisAndRecommendations/DecisionOnRelease/index.js:39]: the switch needs to be on the linkId, the text can be changed. it's better to put the text in a comment then the code will be more readable

File: src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js (hunk 5 above)
Comment (idan-matrix) [reply src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js:271]: this is not the story for this mission

File: src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js (hunk 6 above)
Comment (idan-matrix) [reply src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js:334]: I'll be waiting for 14:30 stand up meeting to discuss about it so I can understand better what you meant

**Pull request merged**